Support Ruby 3.x through 4.0

anitagraham opened the pull request:
Updates refinerycms-i18n to Ruby 3, to match refinery/refinerycms#3518.

Using Mobility 1.3.x

Apart from that most changes are extremely minor.

anitagraham committed:
Try adding workflow actions

anitagraham committed:
Minor editing of documentation
Update mobility to version 1.2.9 supporting Ruby 3.
Update tests to use Ruby 3 syntax

anitagraham committed:
Use matching ruby3 branch of refinerycms while still a work in progress

anitagraham committed:
CI - run on Ruby 3.1

fembem committed:
result is a url, not a page

parndt commented on refinerycms-i18n.gemspec:
```diff
@@ -14,11 +14,11 @@ Gem::Specification.new do |s|
   s.test_files        = `git ls-files -- spec/*`.split("\n")
 
-  s.required_ruby_version = '>= 2.2.9'
+  s.required_ruby_version = '>= 3.1'
```
```suggestion
  s.required_ruby_version = '>= 2.7'
```

parndt committed:
it should still work with Ruby 2.7+

parndt committed:
Update main.yml

parndt committed:
Merge branch 'master' into ruby3

parndt committed:
File.exists? -> File.exist?

parndt commented:
Got CI running. Strangely, only Ruby 3.1 works and not 3.0 or 3.2 🤔

parndt committed:
Update main.yml

halteren commented:
Shouldn't line 29 of /lib/refinery/i18n-filter.rb be changed to

 result**.url** !~ %r{^/(#{Refinery::Core.backend_route}|wymiframe)}

anitagraham committed:
Merge pull request #2 from fembem/around-generate-method-fix

result is a url, not a page

anitagraham commented:
Closed and reopened to trigger new set of checks

anitagraham committed:
refactor(branch ruby3): compatible with refinerycms/ruby3

update version to 5.0.2

override the routing-filter/find_routes as recommended in https://github.com/svenfuchs/routing-filter/pull/87

move complex conditions to methods in i18n-filter

clarify logic in find_or_set_local

remove flag icons as they are no longer used

parndt committed:
Merge branch 'master' into ruby3

parndt committed:
Refactor locale setting logic in find_or_set_locale

parndt committed:
Update mobility dependency version constraint

parndt committed:
Update routing filter overrides for Rails 8+ support

Refactor routing filter overrides for Rails compatibility.

parndt committed:
Update CI workflow to support new Ruby and database versions

parndt committed:
Enhance routing filter for Rails 7 and 8 compatibility

Refactor routing filter for Rails compatibility.

parndt committed:
Fix routing filter for Rails 7.1 and 8.0 compatibility

Updated routing filter compatibility for Rails versions 7.1 and 8.0.

parndt committed:
Update routing filter for Rails 7.1 and 8 compatibility

Refactor routing filter overrides for Rails compatibility.

parndt committed:
Only remove the existing patch if we're overriding it

parndt committed:
Update refinerycms-i18n.gemspec

parndt commented:
Thanks so much @anitagraham
